src/script/sign.h

```diff
@@ -31,6 +31,17 @@ class BaseSignatureCreator {
     virtual bool CreateSig(const SigningProvider& provider, std::vector<unsigned char>& vchSig, const CKeyID& keyid, const CScript& scriptCode, SigVersion sigversion) const =0;
 };
 
+/** A general purpose signature creator. Note that this lacks assumptions about the `sighash`, as the preimage is not known to the signer. */
+class SimpleSignatureCreator : public BaseSignatureCreator
+{
+    SimpleSignatureChecker checker;
+
+public:
+    explicit SimpleSignatureCreator(const uint256& hashIn) : BaseSignatureCreator(), checker(hashIn) {};
```

Comment:
Is there any reason why the constructor of this class calls its base class constructor `BaseSignatureCreator()`, while the constructor of `SimpleSignatureChecker()` doesn't?

Reply:
Nope. Shouldn't change anything in practice, but consistency is nice. Unifying.